homeassistant/helpers/entity_platform.py

```diff
@@ -610,6 +610,7 @@ async def _async_add_entity(  # noqa: C901
                 hidden_by=hidden_by,
                 known_object_ids=self.entities.keys(),
                 has_entity_name=entity.has_entity_name,
+                initial_options=entity.initial_entity_options,
```

Comment:
It's a bit annoying that we are always generating initial entity options since it's only needed once. Can we make this a function? 
```suggestion
                get_initial_options=entity.get_initial_entity_options,
```